fallback/s2a_fallback.go

```diff
@@ -0,0 +1,159 @@
+/*
+ *
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package fallback provides default implementations of fallback options when S2A fails.
+package fallback
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/grpclog"
+	"net"
+)
+
+const (
+	alpnProtoStrH2   = "h2"
+	alpnProtoStrHttp = "http/1.1"
+	defaultHttpsPort = "443"
+)
+
+// FallbackClientHandshake establishes a TLS connection and returns it, plus its auth info.
+// Input:
+//
+//	targetServer: the server attempted with S2A.
+//	conn: the raw tcp connection passed into S2A's ClientHandshake func.
```

Comment:
nit: For clarity, it may be good to say "the tcp connection to the server at address targetServer that was ..."